lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -102,18 +102,24 @@ def scrub(transcript)
 
       private
 
-      def add_extra_data(post, options)
+      NETWORK_TOKENIZATION_CARD_SOURCE = {
+        "apple_pay" => "applepay",
+        "android_pay" => "androidpay",
+        "google_pay" => "paywithgoogle"
+      }
+
+      def add_extra_data(post, payment, options)
         post[:shopperEmail] = options[:shopper_email] if options[:shopper_email]
         post[:shopperIP] = options[:shopper_ip] if options[:shopper_ip]
         post[:shopperReference] = options[:shopper_reference] if options[:shopper_reference]
         post[:fraudOffset] = options[:fraud_offset] if options[:fraud_offset]
-        post[:selectedBrand] = options[:selected_brand] if options[:selected_brand]
+        post[:selectedBrand] = options[:selected_brand] || NETWORK_TOKENIZATION_CARD_SOURCE[payment.source.to_s] if payment.is_a?(NetworkTokenizationCreditCard)
```

Comment:
Is it okay to pass this field empty?

Reply:
I don't think it's possible to be sent as empty, given the conditions... if there's no `options[:selected_brand]` and it's not a `NetworkTokenizationCreditCard`, the field doesn't get added, otherwise a value will be present.